x-pack/plugins/security_solution/public/common/components/navigation/breadcrumbs/index.ts

```diff
@@ -64,158 +71,104 @@ export const useSetBreadcrumbs = () => {
-      ...getUsersBreadcrumbs(
-        spyState,
-        urlStateKeys.reduce(
-          (acc: string[], item: SearchNavTab) => [...acc, getSearch(item, object)],
-          []
-        ),
-        getUrlForApp
-      ),
-    ];
+const getTrailingBreadcrumbsForRoutes = (
+  spyState: RouteSpyState,
+  getSecuritySolutionUrl: GetSecuritySolutionUrl
+): ChromeBreadcrumb[] => {
+  if (isHostsRoutes(spyState)) {
+    return getHostDetailsBreadcrumbs(spyState, getSecuritySolutionUrl);
+  }
+  if (isNetworkRoutes(spyState)) {
+    return getIPDetailsBreadcrumbs(spyState, getSecuritySolutionUrl);
   }
 
-  if (isRulesRoutes(spyState) && object.navTabs) {
-    const tempNav: SearchNavTab = { urlKey: SecurityPageName.rules, isDetailPage: false };
-    let urlStateKeys = [getOr(tempNav, spyState.pageName, object.navTabs)];
-    if (spyState.tabName != null) {
-      urlStateKeys = [...urlStateKeys, getOr(tempNav, spyState.tabName, object.navTabs)];
-    }
-
-    return [
-      siemRootBreadcrumb,
-      ...getDetectionRulesBreadcrumbs(
-        spyState,
-        urlStateKeys.reduce(
-          (acc: string[], item: SearchNavTab) => [...acc, getSearch(item, object)],
-          []
-        ),
-        getUrlForApp
-      ),
-    ];
+  if (isUsersRoutes(spyState)) {
+    return getUsersBreadcrumbs(spyState, getSecuritySolutionUrl);
   }
 
-  if (isCaseRoutes(spyState) && object.navTabs) {
-    return null; // controlled by Cases routes
+  if (isRulesRoutes(spyState)) {
+    return getDetectionRulesBreadcrumbs(spyState, getSecuritySolutionUrl);
   }
 
-  if (isTimelinesRoutes(spyState) && object.navTabs) {
-    const tempNav: SearchNavTab = { urlKey: 'timeline', isDetailPage: false };
-    const urlStateKeys = [getOr(tempNav, spyState.pageName, object.navTabs)];
+  return [];
+};
 
-    return [
-      siemRootBreadcrumb,
-      ...getTimelinesBreadcrumbs(
-        spyState,
-        urlStateKeys.reduce(
-          (acc: string[], item: SearchNavTab) => [...acc, getSearch(item, object)],
-          []
-        ),
-        getUrlForApp
-      ),
-    ];
-  }
+const isNetworkRoutes = (spyState: RouteSpyState): spyState is NetworkRouteSpyState =>
+  spyState.pageName === SecurityPageName.network;
 
-  if (isAdminRoutes(spyState) && object.navTabs) {
-    return [siemRootBreadcrumb, ...getAdminBreadcrumbs(spyState)];
-  }
+const isHostsRoutes = (spyState: RouteSpyState): spyState is HostRouteSpyState =>
+  spyState.pageName === SecurityPageName.hosts;
+
+const isUsersRoutes = (spyState: RouteSpyState): spyState is UsersRouteSpyState =>
+  spyState.pageName === SecurityPageName.users;
+
+const isCaseRoutes = (spyState: RouteSpyState) => spyState.pageName === SecurityPageName.case;
+
+const isAdminRoutes = (spyState: RouteSpyState): spyState is AdministrationRouteSpyState =>
+  spyState.pageName === SecurityPageName.administration;
+
+const isRulesRoutes = (spyState: RouteSpyState): spyState is AdministrationRouteSpyState =>
+  spyState != null &&
```

Comment:
question: why in `isRulesRoutes` present check for null but absent in other isRoute functions?
I can see check for null was present on other functions before these changes